lms/djangoapps/edraak_forus/views.py

```diff
@@ -0,0 +1,291 @@
+                return redirect('dashboard')
+        except InvalidKeyError:
+            return forus_error_redirect(_("Invalid course id"))
+        except Exception as error:  # pylint: disable=broad-except, invalid-name
+            log.exception(error)
+            return forus_error_redirect(_("Could not enroll"))
+
+    def get_registration_form_description(self, request, forus_params):
+        """Retrieve form descriptions from the user API.
+
+        Arguments:
+            request (HttpRequest): The original request, used to retrieve session info.
+            forus_params (dict): forus params values.
+
+        Returns:
+            JSON-serialized registration form descriptions.
+        """
+        form = json.loads(_local_server_get(reverse('forus_v1_reg_api'), request.session))
+
+        for field in form['fields']:
+            field_name = field['name']
+
+            if field_name in POPULATED_FIELDS:
+                field['defaultValue'] = forus_params[field_name]
+
+            if 'username' == field_name:
+                field['defaultValue'] = forus_params['name'].strip()
+
+        return form
+
+    def render_auth_form(self, request, forus_params):
+        """
+        Render the combined login/registration form, defaulting to login
+
+        This relies on the JS to asynchronously load the actual form from
+        the user_api.
+        """
+        initial_mode = 'register'
+
+        # Determine the URL to redirect to following login/registration/third_party_auth
+        redirect_to = get_next_url_for_login_page(request)
+
+        # If we're already logged in, redirect to the dashboard
+        if request.user.is_authenticated():
+            logout(request)
+            delete_logged_in_cookies(request)
+            return redirect(redirect_to)
+
+        # Otherwise, render the combined login/registration page
+        context = {
+            'data': {
+                'login_redirect_url': redirect_to,
+                'initial_mode': initial_mode,
+                'third_party_auth': {},
+                'third_party_auth_hint': '',
+                'platform_name': settings.PLATFORM_NAME,
+
+                # Include form descriptions retrieved from the user API.
+                # We could have the JS client make these requests directly,
+                # but we include them in the initial page load to avoid
+                # the additional round-trip to the server.
+                'registration_form_desc': self.get_registration_form_description(request, forus_params),
+            },
+            'login_redirect_url': redirect_to,  # This gets added to the query string of the "Sign In" button in header
+            'responsive': True,
+            'allow_iframing': True,
+            'disable_courseware_js': True,
+            'disable_footer': True,
+        }
+
+        return render_to_response('edraak_forus/auth.html', context)
+
+    def dispatch(self, request, *args, **kwargs):
```

Comment:
@Salomari1987 I think we can do this later:

 - [Activate forus API lang param with integration tests with edraak_i18n app](https://app.asana.com/0/256944342521670/305314536764578).